src/frontend/src/App.tsx

```diff
@@ -1,26 +1,38 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+import { Home } from "./pages/home/home"
+import { Login } from './pages/login/login';
+import { NewChallenge } from './pages/new-challenge/new-challenge';
+import { ViewChallenge } from './pages/view-challenge/view-challenge';
+
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Redirect
+} from "react-router-dom";
+
+const App: React.FC = () => {
+    return (
+        <div className="App">
+            <Router>
+                    <Switch>
+                        {/* TODO: Whoever ends up implementing the login will need to change these routes to "PrivateRoute" so that
+                            users who have not logged in are unable to access the other Routes.
+                            Some links that might help:
+                            - https://tylermcginnis.com/react-router-protected-routes-authentication/
+                            - https://medium.com/@tomlarge/private-routes-with-react-router-dom-28e9f40c7146 */}
+                        <Route path="/login" children={ <Login /> } />
```

Comment:
```suggestion
                        <Route path="/login">
                            <Login />
                        </Route>
```
I think nesting JSX components looks neater.